### ISSUE TITLE
[Question] 'outdated' not a valid setting for 'build_policy'

### ISSUE THREAD

**Issue by marausch** [conan-io/conan#3892]
Hello, quick question:
Seems that `build_policy = "outdated"` doesn't work and in fact only `missing` and `always` are listed as possible values.
Only the `--build` option to `conan install` accepts that value.
Is this correct? And is it intentional?
TIA

**Comment by memsharded** [comment]
Yes, it is intentional. The ``build_policy`` has implemented the following values so far:

- ``always``: For living on the edge. Having a source method that clones from a branch HEAD, that it is being continously modified, so you want to make sure that the package is built always with the latest code
- ``missing``: For libraries that cannot provide binaries, because they really need to be built on the host machine (e.g. they depend on specific drivers of that machine that are not available otherwise, and the linked artifacts are not relocatable to other machines)

Both are actually very exceptional cases, I would advise using them unless there is very strong evidence that there is no other way. ``build=outdated`` is a very developer specific flow, that shouldn't be used in most normal scenarios, so it seems that it doesn't make sense as default hardcoded in recipes.

**Comment by puetzk** [comment]
Another I'd kind of like to see is build="never", for recipes that are only meant to be used with export-pkg to [distribute pre-built binaries](https://docs.conan.io/en/latest/creating_packages/existing_binaries.html#packaging-pre-built-binaries) (e.g. a library that must be signed, so a normal consumer cannot produce their own working build)

Though you can pretty much achieve that goal just by throwing an exception in `def build(self)`, knowing it up front would let conan install know it doesn't have a viable build plan, instead of starting down the process of building missing things and then hitting one that fails.
